src/main/scala/firrtl/backends/verilog/PadWidthsVerilog.scala

```diff
@@ -0,0 +1,46 @@
+// See LICENSE for license details.
+
+package firrtl.backends.verilog
+
+import firrtl.ir._
+import firrtl.PrimOps._
+import firrtl.Mappers._
+import firrtl.{Dshlw, SystemVerilogEmitter, Transform, VerilogEmitter, bitWidth}
+import firrtl.options.Dependency
+import firrtl.stage.Forms
+import firrtl.stage.TransformManager.TransformDependency
+
+
+/** Adds padding for mul, div, rem, dshl which breaks firrtl width invariance, but is nmeeded to match Verilog semantics */
+object PadWidthsVerilog extends firrtl.passes.Pass {
+
+  override def prerequisites: Seq[TransformDependency] = Forms.MidForm ++
+    Seq( Dependency(firrtl.passes.LowerTypes),
+         Dependency(firrtl.transforms.RemoveReset),
+         Dependency[firrtl.transforms.RemoveWires] )
+
+  override def optionalPrerequisiteOf =
+    Seq( Dependency(firrtl.passes.memlib.VerilogMemDelays),
+      Dependency[SystemVerilogEmitter],
+      Dependency[VerilogEmitter] )
+
+  override def invalidates(a: Transform): Boolean = false
```

Comment:
Since this introduces custom IR nodes and changes width semantics, I'm pretty sure that this invalidates the world (or at least `InferTypes`).

That said, fixing that in this PR may be complicated to do correctly (as it would likely mean changing Verilog forms to not be supersets of higher forms and that may have ripple effects to lots of files).

I think a reasonable compromise is to make `PadWidthsVerilog` `private[firrtl]` until we figure out how to do handle the refactor.

Reply:
I made it package private.

The old version would invalidate `Legalize` and `ConstantProp` and I think (hope) I solved that by manually calling constant prop when adding new `pad` or `bits` nodes.